src/sql/azdata.proposed.d.ts

```diff
@@ -1507,6 +1507,10 @@ declare module 'azdata' {
 		 * Edges corresponding to the children.
 		 */
 		edges: ExecutionPlanEdge[];
+		/**
+		 * Unique internal id given to graph node by ADS.
+		 */
+		id?: string;
```

Comment:
Why is this optional? Shouldn't every node have an ID?

Reply:
is the comment accurate? is the value assigned by mssql extension or ADS core?

Reply:
for internal id used only by ADS, you don't have to expose it here.

Reply:
The id will be given by frontend to all graph nodes and edges.

Reply:
This helps in searching and selecting nodes

Reply:
yes, but does the backend need to know about it? if not, you shouldn't expose it.